agent/engine/containermetadata/types.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package containermetadata
+
+import (
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/agent/api"
+
+	docker "github.com/fsouza/go-dockerclient"
+)
+
+// dockerDummyClient is a wrapper for the docker interface functions we need
+// We use this as a dummy type to be able to pass in engine.DockerClient to
+// our functions without creating import cycles
+type dockerDummyClient interface {
```

Comment:
I cannot use the one in the engine package because we would require importing "engine" and that leads to import cycles.